### PR TITLE
Add recording events in Flutter

### PR DESCRIPTION
**Pull request by kbetl-dlb**

### CODE REVIEW

File: android/src/main/kotlin/io/dolby/comms/sdk/flutter/events/RecordingEventEmitter.kt [hunk 1]
```diff
@@ -0,0 +1,20 @@
...
+class RecordingEventEmitter(eventChannelHandler: EventChannelHandler) : NativeEventEmitter(eventChannelHandler) {
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    fun onRecordingStatusUpdate(event: RecordingStatusUpdatedEvent) {
+        mapOf(
+            "recordingStatus" to "EVENT_${event.recordingStatus}",
+            "conferenceId" to event.conferenceId,
+            "participantId" to event.participantId,
+            "timeStamp" to event.timeStamp
+        ).let {
+            emit("EVENT_${event.recordingStatus}", it)
```
Comment (kdrob-dolby) [inline android/src/main/kotlin/io/dolby/comms/sdk/flutter/events/RecordingEventEmitter.kt:17]: We should stick to one event like native sdks do. Like "EVENT_RECORDING_STATUS_UPDATED".

File: android/src/main/kotlin/io/dolby/comms/sdk/flutter/events/RecordingEventEmitter.kt [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+import com.voxeet.sdk.json.RecordingStatusUpdatedEvent
+import com.voxeet.sdk.models.v1.RecordingStatus
+import org.greenrobot.eventbus.Subscribe
+import org.greenrobot.eventbus.ThreadMode
+
+class RecordingEventEmitter(eventChannelHandler: EventChannelHandler) : NativeEventEmitter(eventChannelHandler) {
+    @Subscribe(threadMode = ThreadMode.MAIN)
+    fun onRecordingStatusUpdate(event: RecordingStatusUpdatedEvent) {
+        mapOf(
+            "recordingStatus" to "EVENT_${event.recordingStatus}",
```
Comment (kdrob-dolby) [inline android/src/main/kotlin/io/dolby/comms/sdk/flutter/events/RecordingEventEmitter.kt:12]: It should be `"recordingStatus" to event.recordingStatus`

File: lib/src/sdk_api/models/events.dart [hunk 3]
```diff
@@ -68,3 +70,26 @@ class StreamsChangeData {
...
+  String conferenceId;
+  String participantId;
+  int timeStamp;
+
+  RecordingStatusUpdate(
+      this.recordingStatus,
+      this.conferenceId,
+      this.participantId,
+      this.timeStamp
+      );
```
Comment (kdrob-dolby) [inline lib/src/sdk_api/models/events.dart:85]: Formatting
```suggestion
    RecordingStatusUpdate(
    this.recordingStatus,
    this.conferenceId,
    this.participantId,
    this.timeStamp,
  );
```

File: lib/src/sdk_api/models/events.dart [hunk 4]
```diff
@@ -68,3 +70,26 @@ class StreamsChangeData {
...
+
+  static RecordingStatusUpdate fromMap(Map<Object?, Object?> data) {
+    RecordingServiceEventNames recordingStatus = RecordingServiceEventNames.valueOf(data["recordingStatus"] as String);
+    String conferenceId = data["conferenceId"] as String;
+    String participantId = data["participantId"] as String;
+    int timeStamp = data["timeStamp"] as int;
+    return RecordingStatusUpdate(recordingStatus, conferenceId, participantId, timeStamp);
+  }
+
```
Comment (kdrob-dolby) [inline lib/src/sdk_api/models/events.dart:94]: Formatting
```suggestion
  static RecordingStatusUpdate fromMap(Map<Object?, Object?> data) {
    RecordingServiceEventNames recordingStatus =
        RecordingServiceEventNames.valueOf(data["recordingStatus"] as String);
    String conferenceId = data["conferenceId"] as String;
    String participantId = data["participantId"] as String;
    int timeStamp = data["timeStamp"] as int;
    return RecordingStatusUpdate(
      recordingStatus,
      conferenceId,
      participantId,
      timeStamp,
    );
  }
```

**Review by kdrob-dolby — CHANGES_REQUESTED** [review]

File: android/src/main/kotlin/io/dolby/comms/sdk/flutter/events/RecordingEventEmitter.kt (hunk 1 above)
Comment (kbetl-dlb) [reply android/src/main/kotlin/io/dolby/comms/sdk/flutter/events/RecordingEventEmitter.kt]: changed

File: android/src/main/kotlin/io/dolby/comms/sdk/flutter/events/RecordingEventEmitter.kt (hunk 2 above)
Comment (kbetl-dlb) [reply android/src/main/kotlin/io/dolby/comms/sdk/flutter/events/RecordingEventEmitter.kt]: changed

File: lib/src/sdk_api/models/events.dart (hunk 3 above)
Comment (kbetl-dlb) [reply lib/src/sdk_api/models/events.dart]: changed

File: lib/src/sdk_api/models/events.dart (hunk 4 above)
Comment (kbetl-dlb) [reply lib/src/sdk_api/models/events.dart]: changed

File: lib/src/sdk_api/recording_service.dart [hunk 5]
```diff
@@ -32,4 +40,16 @@ class RecordingService {
...
   }
+
+  Stream<Event<RecordingServiceEventNames,
+          RecordingStatusUpdate>> onRecordingStatusUpdate() {
+    return _eventStream.addListener(
+        [RecordingServiceEventNames.recordingStatusUpdate.value]).map((map) {
+      final event = map as Map<Object?, Object?>;
+      final key = RecordingServiceEventNames.valueOf(event["key"] as String);
+      final data = event["body"] as Map<Object?, Object?>;
+      dev.log("[KB]: onRecordingStatusUpdate: event ${data["recordingStatus"]}");
```
Comment (kdrob-dolby) [inline lib/src/sdk_api/recording_service.dart:51]:
```suggestion
```

File: lib/src/sdk_api/recording_service.dart [hunk 6]
```diff
@@ -1,6 +1,9 @@
+import 'package:dolbyio_comms_sdk_flutter/dolbyio_comms_sdk_flutter.dart';
+import 'package:dolbyio_comms_sdk_flutter/src/dolbyio_comms_sdk_native_events.dart';
+
 import '../dolbyio_comms_sdk_flutter_platform_interface.dart';
 import '../mapper/mapper.dart';
-import 'models/recording.dart';
+import "dart:developer" as dev;
```
Comment (kdrob-dolby) [inline lib/src/sdk_api/recording_service.dart:6]: Unused import
```suggestion
```

Comment (kbetl-dlb) [reply lib/src/sdk_api/recording_service.dart]: removed

**Review by kdrob-dolby — APPROVED** [review]

**Pull request merged**